### PR TITLE
[SPARK-12523][YARN]Support long-running of the Spark On HBase and hive meta store.

### PR DESCRIPTION
**Pull request by SaintBacchus**
Obtain the hive metastore and hbase token as well as hdfs token in DelegationToeknRenewer to supoort long-running application of spark on hbase or thriftserver.

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala [hunk 1]
```diff
@@ -22,6 +22,8 @@ import java.nio.charset.StandardCharsets.UTF_8
 import java.util.regex.Matcher
 import java.util.regex.Pattern
 
+import org.apache.spark.deploy.yarn.Client._
```
Comment (tgravescs) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala]: why are we importing this?

**Comment by tgravescs** [comment]
looks pretty good, what testing have you done on this?

**Comment by SaintBacchus** [comment]
@tgravescs we supported to run `spark on hbase` beyond 7 days. It work well but I did not test with hive `metastore` and this is the similar case.

**Comment by tgravescs** [comment]
ok can you remove the extra import or tell me what its needed for

**Comment by tgravescs** [comment]
Jenkins, test this please

**Comment by tgravescs** [comment]
+1

**Pull request closed without merging**